src/Http/Middleware/ShareInertiaData.php

```diff
@@ -16,16 +16,19 @@ class ShareInertiaData
      */
     public function handle($request, $next)
     {
-        Inertia::share(array_filter([
-            'socialstream' => function () use ($request) {
-                return [
-                    'show' => Socialstream::show(),
-                    'providers' => Socialstream::providers(),
-                    'hasPassword' => ! is_null($request->user()->password),
-                    'connectedAccounts' => $request->user()->connectedAccounts,
-                ];
-            },
-        ]));
+        $user = $request->user();
+        if ($user) {
+            Inertia::share(array_filter([
+                'socialstream' => function () use ($user) {
+                    return [
+                        'show' => Socialstream::show(),
+                        'providers' => Socialstream::providers(),
+                        'hasPassword' =>  ! is_null($user->password),
```

Comment:
@luiyongsheng rather than wrapping this in a conditional, can we always expose this data by adjusting the requirements for these flags to use the `&&` operator, like this?

```suggestion
                        'hasPassword' =>  $request->user() && ! is_null($request->user()->password),
```

Reply:
Sure! This is much elegant.

Reply:
Anyway, I still wonder, if a site with huge daily traffic, do these extra bytes of data passed from the backend sums up to a significant amount?